optimum/exporters/onnx/model_configs.py

```diff
@@ -731,3 +731,18 @@ def generate_dummy_inputs(self, framework: str = "pt"):
 class WhisperOnnxConfig(AudioToTextOnnxConfig):
     NORMALIZED_CONFIG_CLASS = NormalizedSeq2SeqConfig
     ATOL_FOR_VALIDATION = 1e-3
+
+    def _create_dummy_input_generator_classes(self, **kwargs) -> List["DummyInputGenerator"]:
+        dummy_inputs_generators = super()._create_dummy_input_generator_classes(**kwargs)
+
+        if self._behavior is ConfigBehavior.MONOLITH:
```

Comment:
why is it in the monolith case? In https://github.com/huggingface/optimum/pull/581/files I think the issue was not only with monolith

Reply:
Edit:

Before this PR, have no issue with
```
python3 -m optimum.exporters.onnx --model openai/whisper-tiny.en --task speech2seq-lm-with-past --for-ort out_whisper
python3 -m optimum.exporters.onnx --model openai/whisper-tiny.en --task speech2seq-lm --for-ort out_whisper
python3 -m optimum.exporters.onnx --model openai/whisper-tiny.en --task speech2seq-lm out_whisper
```

but have an issue with
```
python3 -m optimum.exporters.onnx --model openai/whisper-tiny.en --task speech2seq-lm-with-past out_whisper
```

which raises:

```
An error occured with error [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Invalid Feed Input Name:past_key_values.3.encoder.value.
```